tests/solutions/test_models.py

```diff
@@ -1,101 +1,61 @@
+        mock.solution.task.slug = "here-be-dragons"
+        mock.solution.author = "Mr. Mustermann"
+        mock.solution.id = "123456"
 
-    def test_failure_means_not_passed(self):
-        """Test if non-zero rc marks the solution as not passed."""
-        self.solution.do_check(self.create_mock_checker(
-            ResultTuple(1, "OUT", "ERR", 0.2, dict())
-        ))
-        results = self.solution.checkerresult_set.all()
-        self.assertEqual(results.count(), 1)
-        self.assertFalse(self.solution.passed)
+        upload_path = get_upload_path(mock, "Fibonacci.java")
+        self.assertTrue(upload_path.startswith("solutions"))
+        self.assertTrue(upload_path.endswith("Fibonacci.java"))
 
-    def test_solution_input_is_used(self):
-        """Test if TaskSolutionFiles are passed to the Checker."""
-        checker = self.create_mock_checker(
-            ResultTuple(0, "OUT", "ERR", 1.2, dict())
-        )
-        self.solution.do_check(checker)
-        checker.check_task.assert_called_with(
-            self.task_defaults["name"],
-            path.dirname(self.solutionfile.file.path)
-        )
-
-    def test_get_upload_path(self):
-        solution = self.create_solution()
-        tsf = self.create_solution_file(solution=solution)
-        self.assertRegex(
-            models.get_upload_path(tsf, tsf.filename),
-            (r"solutions/chuck_norris/published-task/"
-             "[\d]{4}/[\d]{2}/[\d]{2}/[\d]{2}_[\d]{1,2}_[\d]+/HelloWorld.java")
-        )
+        tokens = upload_path.split("/")
+        for token in ["1970", "here-be-dragons", "123456"]:
+            self.assertTrue(token in tokens)
 
     def test_solution_default_status(self):
-        solution = self.create_solution()
-        self.assertFalse(solution.checkerresult_set.exists())
-        self.assertEqual(solution.status(), "pending")
+        """Verify that the default status of a solution is pending"""
 
-    def test_solution_lost_status(self):
-        solution = self.create_solution()
-        mocked_time = timezone.now() + timezone.timedelta(minutes=6)
-        with mock.patch("django.utils.timezone.now", return_value=mocked_time):
-            self.assertFalse(solution.checkerresult_set.exists())
-            self.assertEqual(solution.status(), "lost")
+        # Both failed_solution and passed_solution were not checked yet
+        # so their status should be pending
+        self.assertTrue(self.failed_solution.status() == "pending")
+        self.assertTrue(self.passed_solution.status() == "pending")
 
-    def test_solution_failure_status(self):
-        solution = self.create_solution()
-        solution.checkerresult_set.add(CheckerResult(), bulk=False)
-        self.assertEqual(solution.status(), "failure")
+    def test_solution_lost_status(self):
+        """
+        Verify that a solution is lost when there was
+        no background queue for a reasonable amount
+        of time
```

Comment:
* Verify that a solution is lost when it was not processed in a reasonable amount of time.
* End with a dot. ;-)